content/flux/v0.x/query-data/csv.md

```diff
@@ -0,0 +1,268 @@
+
+| dataset     | metric   | sensorID | timestamp           | value |
+| :---------- | :------- | :------- | :------------------ | ----: |
+| air-sensors | humidity | TLM0200  | 1627049400000000000 | 35.64 |
+| air-sensors | humidity | TLM0200  | 1627049700000000000 | 35.67 |
+
+| dataset     | metric      | sensorID | timestamp           | value |
+| :---------- | :---------- | :------- | :------------------ | ----: |
+| air-sensors | temperature | TLM0100  | 1627049400000000000 | 71.84 |
+| air-sensors | temperature | TLM0100  | 1627049700000000000 | 71.87 |
+
+| dataset     | metric      | sensorID | timestamp           | value |
+| :---------- | :---------- | :------- | :------------------ | ----: |
+| air-sensors | temperature | TLM0200  | 1627049400000000000 | 74.10 |
+| air-sensors | temperature | TLM0200  | 1627049700000000000 | 74.17 |
+
+---
+
+### Query a raw CSV string
+Use the `csv` parameter to specify the raw CSV string to query.
+Set the `mode` parameter to `raw`.
+
+#### Query
+```js
+import "csv"
+
+csvData = "
+dataset,metric,sensorID,timestamp,value
+air-sensors,humidity,TLM0100,1627049400000000000,34.79
+air-sensors,humidity,TLM0100,1627049700000000000,34.65
+air-sensors,humidity,TLM0200,1627049400000000000,35.64
+air-sensors,humidity,TLM0200,1627049700000000000,35.67
+air-sensors,temperature,TLM0100,1627049400000000000,71.84
+air-sensors,temperature,TLM0100,1627049700000000000,71.87
+air-sensors,temperature,TLM0200,1627049400000000000,74.10
+air-sensors,temperature,TLM0200,1627049700000000000,74.17
+"
+
+csv.from(csv: csvData, mode: "raw")
+```
+
+#### Results
+{{% note %}}
+When using the **raw** CSV parsing mode, all columns values are strings.
+{{% /note %}}
+
+| dataset     | metric      | sensorID | timestamp           | value |
+| :---------- | :---------- | :------- | :------------------ | ----: |
+| air-sensors | humidity    | TLM0100  | 1627049400000000000 | 34.79 |
+| air-sensors | humidity    | TLM0100  | 1627049700000000000 | 34.65 |
+| air-sensors | humidity    | TLM0200  | 1627049400000000000 | 35.64 |
+| air-sensors | humidity    | TLM0200  | 1627049700000000000 | 35.67 |
+| air-sensors | temperature | TLM0100  | 1627049400000000000 | 71.84 |
+| air-sensors | temperature | TLM0100  | 1627049700000000000 | 71.87 |
+| air-sensors | temperature | TLM0200  | 1627049400000000000 | 74.10 |
+| air-sensors | temperature | TLM0200  | 1627049700000000000 | 74.17 |
+
+---
+
+### Query CSV data from a file
+Use `csv.from()` and the `file` parameter to query CSV data from a file.
+
+{{% note %}}
+#### Flux must have access to the file system
+To query CSV data from a file, Flux must have access to the filesystem.
+If Flux does not have access to the file system, the query will return an error
+simlir to:
+
+```
+failed to read file: filesystem service is uninitialized
+```
+If using InfluxDB Cloud or InfluxDB OSS, the Flux process **does not** have 
+access to the filesystem.
```

Comment:
~~~suggestion
If Flux does not have access to the file system, the query will return an error
similar to:

```
failed to read file: filesystem service is uninitialized
```

If using InfluxDB Cloud or InfluxDB OSS, the Flux process **does not** have 
access to the filesystem.
~~~